io/src/openni_camera/openni_driver.cpp

```diff
@@ -557,12 +557,7 @@ openni_wrapper::OpenNIDriver::DeviceContext::DeviceContext (const xn::NodeInfo&
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 openni_wrapper::OpenNIDriver::DeviceContext::DeviceContext (const DeviceContext& other)
-: device_node (other.device_node)
-, image_node (other.image_node)
-, depth_node (other.depth_node)
-, ir_node (other.ir_node)
-, device (other.device)
-{
-}
+ 
+= default;
```

Comment:
Will the default behavior set all these fields?

Reply:
Yes; unlike other linters, `clang-tidy` leverages the clang compiler to perform lexical analysis, so I tend to trust its recommendations. (@kunaltyagi can probably comment about the `pcl::weak_ptr` copy construction behavior...)